### PR TITLE
feat: Require Swift DocC only when env var is set

### PR DESCRIPTION
**Pull request by jbelkins**
## Description of changes
Removes DocC as a SDK dependency unless the env var `AWS_SWIFT_SDK_ENABLE_DOCC` is set, since it is essentially unused unless documentation is actually being generated.  There is generally not a reason to enable it by default for customers.

`AWS_SWIFT_SDK_ENABLE_DOCC` is set for the "generate documentation" job on Github, where DocC is actually used.

## New/existing dependencies impact assessment, if applicable
No new dependencies were added to this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dayaffe — APPROVED** [review]

**Review by sichanyoo — APPROVED** [review]
There's a swiftlint warning that got past review from some other PR. Address before merge if possible

**Pull request merged**
